arwen/contexts/storage_test.go

```diff
@@ -373,86 +407,150 @@ func TestStorageContext_StorageProtection(t *testing.T) {
 	}
 	bcHook := &contextmock.BlockchainHookStub{}
 
-	storageContext, _ := NewStorageContext(host, bcHook, elrondReservedTestPrefix)
-	storageContext.SetAddress(address)
+	storageContextInstance, _ := NewStorageContext(host, bcHook, elrondReservedTestPrefix)
+	storageContextInstance.SetAddress(address)
 
 	key := []byte(arwen.ProtectedStoragePrefix + "something")
 	value := []byte("data")
 
-	storageStatus, err := storageContext.SetStorage(key, value)
+	storageStatus, err := storageContextInstance.SetStorage(key, value)
 	require.Equal(t, arwen.StorageUnchanged, storageStatus)
 	require.True(t, errors.Is(err, arwen.ErrCannotWriteProtectedKey))
-	require.Len(t, storageContext.GetStorageUpdates(address), 0)
+	require.Len(t, storageContextInstance.GetStorageUpdates(address), 0)
 
-	storageContext.disableStorageProtection()
-	storageStatus, err = storageContext.SetStorage(key, value)
+	storageContextInstance.disableStorageProtection()
+	storageStatus, err = storageContextInstance.SetStorage(key, value)
 	require.Nil(t, err)
 	require.Equal(t, arwen.StorageAdded, storageStatus)
-	require.Len(t, storageContext.GetStorageUpdates(address), 1)
+	require.Len(t, storageContextInstance.GetStorageUpdates(address), 1)
 
-	storageContext.enableStorageProtection()
-	storageStatus, err = storageContext.SetStorage(key, value)
+	storageContextInstance.enableStorageProtection()
+	storageStatus, err = storageContextInstance.SetStorage(key, value)
 	require.Equal(t, arwen.StorageUnchanged, storageStatus)
 	require.True(t, errors.Is(err, arwen.ErrCannotWriteProtectedKey))
-	require.Len(t, storageContext.GetStorageUpdates(address), 1)
+	require.Len(t, storageContextInstance.GetStorageUpdates(address), 1)
 }
 
 func TestStorageContext_GetStorageFromAddress(t *testing.T) {
 	t.Parallel()
 
-	scAddress := []byte("account")
-	mockOutput := &contextmock.OutputContextMock{}
-	account := mockOutput.NewVMOutputAccount(scAddress)
-	mockOutput.OutputAccountMock = account
-	mockOutput.OutputAccountIsNew = false
+	t.Run("blockchain hook errors", func(t *testing.T) {
+		t.Parallel()
 
-	mockRuntime := &contextmock.RuntimeContextMock{}
-	mockMetering := &contextmock.MeteringContextMock{}
-	mockMetering.SetGasSchedule(config.MakeGasMapForTests())
-	mockMetering.BlockGasLimitMock = uint64(15000)
+		scAddress := []byte("account")
```

Comment:
`scAddress` appears common to both subtests, could be moved outside of them, within the test function.

Reply:
done